File: weather-demo-app-apple/Weather/UI/Welcome/WelcomeViewModel.swift
//
//  WelcomeViewModel.swift
//  Weather
//
//  Created by Onur on 4/11/26.
//

import Foundation
import Combine

@MainActor
final class WelcomeViewModel: ObservableObject {
    @Published private(set) var hasSeenWelcome: Bool

    private let repository: WeatherDataSource

    init(repository: WeatherDataSource) {
        self.repository = repository
        self.hasSeenWelcome = repository.hasSeenWelcome()
    }

    func continueToApp(onFinished: () -> Void) {
        repository.setWelcomeSeen(true)
        hasSeenWelcome = true
        onFinished()
    }
}

File: weather-demo-app-android/app/src/main/java/com/demo/weather/ui/welcome/WelcomeViewModel.kt
package com.demo.weather.ui.welcome

import androidx.lifecycle.ViewModel
import com.demo.weather.data.WeatherDataSource
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow

class WelcomeViewModel(
    private val repository: WeatherDataSource
) : ViewModel() {

    private val _hasSeenWelcome = MutableStateFlow(repository.hasSeenWelcome())
    val hasSeenWelcome: StateFlow<Boolean> = _hasSeenWelcome.asStateFlow()

    fun continueToApp(onFinished: () -> Unit) {
        repository.setWelcomeSeen(true)
        _hasSeenWelcome.value = true
        onFinished()
    }
}